src/data_generators/synthetic_data_generator.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates a DataDesign from DataDesignParameters."""
+
+from absl import app
+from absl import flags
+import math
+from typing import List
+from wfa_planning_evaluation_framework.data_generators.data_design import DataDesign
+from wfa_planning_evaluation_framework.data_generators.data_set import DataSet
+from wfa_planning_evaluation_framework.data_generators.data_set_parameters import (
+    DataSetParameters, GeneratorParameters)
+from wfa_planning_evaluation_framework.data_generators.pricing_generator import PricingGenerator
+from wfa_planning_evaluation_framework.data_generators.fixed_price_generator import FixedPriceGenerator
+from wfa_planning_evaluation_framework.data_generators.impression_generator import ImpressionGenerator
+from wfa_planning_evaluation_framework.data_generators.homogeneous_impression_generator import HomogeneousImpressionGenerator
+from wfa_planning_evaluation_framework.data_generators.publisher_data import PublisherData
+from wfa_planning_evaluation_framework.data_generators.test_synthetic_data_design_config import TestSyntheticDataDesignConfig
+from wfa_planning_evaluation_framework.data_generators.synthetic_data_design_config import SyntheticDataDesignConfig
+from numpy.random import RandomState
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('output_folder', 'TestDataDesign', 'Output Folder.')
+flags.DEFINE_string('data_design_config', 'TestConfig', 'Data Desgin Config.')
+
+name_to_config_dict = {'test': TestSyntheticDataDesignConfig}
+
+
+class SyntheticDataGenerator():
+  """Generates a DataDesign with synthetic data derived from parameters.
+
+    This class translates a DataDesignParameters object to a DataDesign by
+    constructing the underlying objects for a DataSet and duplicating the
+    DataSet with different random seeds.
+    """
+
+  def __init__(self, output_folder: str, config: SyntheticDataDesignConfig):
+    self._config = config
+    self._output_folder = output_folder
+
+  def __call__(self) -> DataDesign:
+    data_design = DataDesign(dirpath=self._output_folder)
+    data_set_parameters_list = self._config.get_data_set_params_list()
```

Comment:
Combine this line and the next?  E.g.,
  for data_set_parameters in self._config.get_data_set_params_list():

Reply:
ah thanks